src/containers/Client.tsx

```diff
@@ -170,75 +230,152 @@ export const Client = ({
+                    >
+                      <VscClose size={20} />
+                    </button>
+                  )}
+                </div>
+              </div>
             ) : (
+              <div>
+                <input
+                  type="text"
+                  placeholder="Type here"
+                  className="input input-bordered w-full max-w-xs"
+                  onChange={(e) => {
+                    setTokenInput(e.target.value);
+                  }}
+                />
+                <button className="btn btn-sm m-2 btn-success" onClick={() => setToken(tokenInput)}>
+                  Save token
+                </button>
+              </div>
+            )}
+          </div>
+
+          <div className="flex flex-row flex-wrap items-start content-start justify-between">
+            <div className="overflow-auto flex-wrap w-full">
               <button
-                disabled={fullState.status !== "joined"}
-                className="btn btn-sm btn-error m-2"
+                className="btn btn-sm m-2"
                 onClick={() => {
-                  if (!trackId) return;
-                  api?.removeTrack(trackId);
-                  setTrackId(null);
+                  setShow(!show);
                 }}
               >
-                Remove track
+                {show ? "Hide client state " : "Show client state "}
               </button>
-            )}
-            <button
-              className="btn btn-sm m-2"
-              onClick={() => {
-                setShow(!show);
-              }}
-            >
-              {show ? "Hide" : "Show"}
-            </button>
-          </div>
-          {Object.values(fullState.local?.tracks || {}).map(({ trackId, stream }) => (
-            <div key={trackId} className="w-40">
-              {stream && <VideoPlayer stream={stream} />}
+              {show && <JsonComponent state={fullState} />}
             </div>
-          ))}
+          </div>
         </div>
-        {show && <JsonComponent state={fullState} />}
+      </div>
+      {tracksId.map((trackId) => (
+        <div key={trackId?.id || "nope"}>
+          {trackId && (
+            <StreamedTrackCard
+              trackInfo={trackId}
+              tracksId={tracksId}
+              setTracksId={setTracksId}
+              allTracks={fullState?.local?.tracks || {}}
+              trackMetadata={trackMetadata || DEFAULT_TRACK_METADATA}
+              removeTrack={(trackId) => {
+                if (!trackId) return;
+                activeStreams?.[trackId]?.stream?.getTracks().forEach((track) => {
+                  track.stop();
+                });
+                api?.removeTrack(trackId);
+                activeOutgoingStreams
+                  .get(trackId)
+                  ?.getTracks()
+                  .forEach((track) => track.stop());
+                setActiveOutgoingStreams((prev) => {
+                  prev.delete(trackId);
+                  return prev;
```

Comment:
I think, it's not a good practice to mutate state object. You should make copy of `prev`